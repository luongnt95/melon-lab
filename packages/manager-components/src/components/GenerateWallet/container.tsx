import { compose, withHandlers, withState } from 'recompose';
import GenerateWallet from './index';

const enhance = compose(
  withState('page', 'setPage', props => {
    return props.page ? props.page : 0;
  }),
  withHandlers({
    nextPage: props => () => {
      props.setPage(props.page + 1);
    },
    prevPage: props => () => {
      props.setPage(props.page - 1);
    },
  }),
);

export default enhance(GenerateWallet);
