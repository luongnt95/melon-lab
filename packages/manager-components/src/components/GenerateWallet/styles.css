.generate-wallet {
  max-width: 400px;
  margin: 0 auto var(--space-xl) auto;
}

.generate-wallet__warning {
  padding: var(--space-m);
  background-color: var(--color-warning);
  border: 1px solid color-mod(var(--color-warning) b(10%));
}

.generate-wallet__mnemonic {
  font-size: var(--font-size-xl);
  font-family: monospace;
  font-weight: var(--font-weight-bold);
  margin-top: var(--space-m);
  background-color: var(--color-secondary);
  padding: var(--space-l) var(--space-xl);
}

.generate-wallet__loading {
  margin: var(--space-s) auto;
  text-align: center;
}

.generate-wallet__loading-spinner {
  width: 50px;
  margin: 0 auto;
}
