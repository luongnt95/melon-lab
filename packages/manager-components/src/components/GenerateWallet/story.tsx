import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import React from 'react';
import GenerateWallet from './container';

const data = {
  generatedMnemonic:
    'next glass shell collect erupt image drive tumble motor gym glove inside',
  restore: action('restore'),
  onSubmitMnemonic: action('onSubmit'),
  restoreWallet: {
    initialValues: {
      mnemonic: '',
    },
  },
  passwordForm: {
    onSubmit: action('onSubmit'),
    initialValues: {
      password: 'abc',
    },
  },
};

storiesOf('Components|Generate Wallet', module).add('Default', () => {
  return <GenerateWallet {...data} />;
});
