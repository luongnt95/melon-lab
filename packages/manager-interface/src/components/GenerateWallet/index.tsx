import GenerateWallet from '@melonproject/manager-components/components/GenerateWallet/container';

export default GenerateWallet;
