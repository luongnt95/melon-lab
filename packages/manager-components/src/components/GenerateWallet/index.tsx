import React, { StatelessComponent, Fragment } from 'react';
import Button from '~/blocks/Button';
import Form from '~/blocks/Form';
import Input from '~/blocks/Input';
import Spinner from '~/blocks/Spinner';

import styles from './styles.css';

interface FormValues {
  mnemonic: string;
  password: string;
}
export interface GenerateWalletProps {
  mnemonic: string;
  loading?: boolean;
  handleBlur?: () => void;
  handleSubmit?: () => void;
  handleChange?: () => void;
  touched?: any;
  values: FormValues;
  errors?: any;
  showForm: boolean;
  setShowForm: (state) => void;
}

export const GenerateWallet: StatelessComponent<GenerateWalletProps> = ({
  mnemonic,
  loading,
  handleSubmit,
  handleChange,
  handleBlur,
  values,
  errors,
  touched,
  showForm,
  setShowForm,
}) => (
  <div className="generate-wallet">
    <style jsx>{styles}</style>
    <h3>Generate Account</h3>
    {loading ? (
      <div className="generate-wallet__loading">
        <div className="generate-wallet__loading-spinner">
          <Spinner icon size="small" />
        </div>
        {!showForm && <Fragment>Generating Wallet...</Fragment>}
      </div>
    ) : (
      <Fragment>
        {!showForm ? (
          <Fragment>
            <p>
              The standard bip39 is used to generate a mnemonic phrase, from
              which your wallet will be cryptographically derived.
            </p>
            <p>
              Please write down the following mnemonic and store it in a safe
              place! If you loose your mnemonic you will not be able to access
              your fund again. If someone else gets a copy of this, they can
              take over your wallet & fund and steal your price!
            </p>
            <p className="generate-wallet__mnemonic">{mnemonic}</p>

            <p>
              <Button style="secondary" onClick={() => setShowForm(true)}>
                I have written down the mnemonic
              </Button>
            </p>
          </Fragment>
        ) : (
          <Form onSubmit={handleSubmit}>
            <div className="generate-wallet__input">
              <Input
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.mnemonic}
                required={true}
                name="mnemonic"
                type="text"
                placeholder="Repeat mnemonic"
                error={touched.mnemonic && errors.mnemonic}
              />
            </div>
            <div className="generate-wallet__input">
              <Input
                maxlength={64}
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.password}
                required={true}
                name="password"
                type="password"
                placeholder="Password"
                error={touched.password && errors.password}
              />
            </div>
            <Button type="submit" style="secondary">
              Generate
            </Button>
          </Form>
        )}
      </Fragment>
    )}
  </div>
);

export default GenerateWallet;
