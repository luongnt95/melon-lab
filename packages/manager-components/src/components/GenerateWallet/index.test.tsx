import React from 'react';
import GenerateWallet from './index';

const data = {
  restore: () => null,
  mnemonic: 'lorem ipsum',
};

describe('GenerateWallet', () => {
  const defaultElement = <GenerateWallet {...data} />;
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(defaultElement);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
