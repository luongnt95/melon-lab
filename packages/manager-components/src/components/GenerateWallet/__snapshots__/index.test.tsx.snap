// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenerateWallet should render correctly 1`] = `
<withHandlers(Component)
  generatedMnemonic="next glass shell collect erupt image drive tumble motor gym glove inside"
  onSubmitMnemonic={[MockFunction]}
  onSubmitPassword={[MockFunction]}
  page={0}
  passwordForm={
    Object {
      "initialValues": Object {
        "password": "abc",
      },
    }
  }
  restoreWallet={
    Object {
      "initialValues": Object {
        "mnemonic": "",
      },
    }
  }
  setPage={[Function]}
/>
`;
